contracts/staking.sol

```diff
@@ -38,8 +36,21 @@ contract Staking {
         return _validators;
     }
 
-    // public function
+    // public functions
+    receive() external payable {
+        _stake();
+    }
```

Comment:
What do you think of adding `fallback()` method as well?
In case someone sends ethers and the data (to a non-existing function, for example). In that case, we can also call `_stake()`.

[Ref](https://blog.soliditylang.org/2020/03/26/fallback-receive-split/)

Reply:
@Vuksan 
I'm afraid I think you can't call the method that isn't defined in the contract, EVM throws error in this case. The method without payable (e.g. `unstake`) rejects transaction with value.

I saw the reference you shared and found the sentence like:

| receive() external payable — for empty calldata (and any value)

Reply:
I fear that implementing `fallback` (since it won't throw an error during execution) might lead users to think they have integrated well when in fact they haven't => thus when we extend the functionality of staking, they will only then realize they've been doing it wrong all along.

Let's keep erroring if the calldata is wrong, and simply revert in this case 👍 @Kourin1996